### PR TITLE
Document credentials over non-TLS connections

### PR DESCRIPTION
**Pull request by jcreixell**
#### PR Description

  - Several users have fallen into this while attempting to configure otlp in internal networking environments, so we add a clarifying note to the documentation.

#### Which issue(s) this PR fixes

  - Fixes https://github.com/grafana/agent/issues/3686

### CODE REVIEW

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 1]
```diff
@@ -107,6 +107,11 @@ server.
 
 {{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" >}}
 
+> **NOTE**: `otelcol.exporter.otlp` uses gRPC, which disallows sending sensitive credentials (like `auth`) over insecure channels for security reasons.
+> If you need to send sensitive credentials over non-TLS connections (via the `insecure` argument), you may want to use [otelcol.exporter.otlphttp][] instead.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:111]:
```suggestion
> If you need to send sensitive credentials over non-TLS connections (via the `insecure` argument), you can use [otelcol.exporter.otlphttp][] instead.
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md [hunk 2]
```diff
@@ -107,6 +107,11 @@ server.
 
 {{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" >}}
 
+> **NOTE**: `otelcol.exporter.otlp` uses gRPC, which disallows sending sensitive credentials (like `auth`) over insecure channels for security reasons.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/otelcol.exporter.otlp.md:110]:
```suggestion
> **NOTE**: `otelcol.exporter.otlp` uses gRPC, which does not allow you to send sensitive credentials (like `auth`) over insecure channels.
```

File: docs/sources/flow/reference/components/otelcol.exporter.otlp.md (hunk 1 above)
Comment (ptodev) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:111]:
```suggestion
> Sending sensitive credentials over insecure non-TLS connections is supported by non-gRPC exporters such as [otelcol.exporter.otlphttp][.
```
What about this wording?

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:111]: From a language standpoint, that's good. It's just a clear statement without giving specific direction (which seems to be more appropriate here).

Comment (jcreixell) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:111]: I am fine with this, @rfratto WDYT?

Comment (rfratto) [reply docs/sources/flow/reference/components/otelcol.exporter.otlp.md:111]: SGTM 👍

**Review by rfratto — APPROVED** [review]

**Comment by rfratto** [comment]
Should this get the backport-v0.33 label to be backported to live docs?

**Pull request merged**